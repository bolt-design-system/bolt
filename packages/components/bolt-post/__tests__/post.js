import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
  vrtDefaultConfig,
} from '../../../testing/testing-helpers';
import schema from '../post.schema';
const { disabled } = schema.properties;
const timeout = 90000;

describe('Post', () => {
  let page;

  afterAll(async () => {
    await stopServer();
    await page.close();
  });

  beforeEach(async () => {
    await page.evaluate(() => {
      document.body.innerHTML = '';
    });
  }, timeout);

  beforeAll(async () => {
    page = await global.__BROWSER__.newPage();
    await page.goto('http://127.0.0.1:4444/', {
      timeout: 0,
    });
  }, timeout);

  test('basic usage', async () => {
    const results = await render('@bolt-components-post/post.twig', {
      headline: {
        text: 'This is a post',
        tag: 'h3',
        size: 'xlarge',
        link_attributes: {
          href: 'https://pega.com',
        },
      },
      meta_items: [
        'Posted 8 hours 15 minutes ago',
        'Last activity: 2 minutes ago',
      ],
      status: {
        solved: true,
        solved_label: 'custom solved label',
        locked: true,
        locked_label: 'custom locked label',
        replies: '33',
        views: '123',
        favorited: true,
        subscribed: true,
      },
      show_actions: {
        reply: true,
        favorite: true,
        subscribe: true,
      },
      more_info: 'This is the more information',
      warning: 'This is a warning message',
      attributes: {
        'data-foo': 'bar',
      },
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
