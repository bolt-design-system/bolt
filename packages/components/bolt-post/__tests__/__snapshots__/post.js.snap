// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post basic usage 1`] = `
<article data-foo="bar"
         class="c-bolt-post js-test-target"
>
  <div class="c-bolt-post__content">
    <div class="c-bolt-post-info">
      <div class="c-bolt-post-info__heading">
        <h3 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
          <a href="https://pega.com"
             class="e-bolt-text-link e-bolt-text-link--reversed-underline e-bolt-text-link--expand-click-target"
          >
            This is a post
          </a>
        </h3>
      </div>
      <ul class="c-bolt-post-info__meta">
        <li class="c-bolt-post-info__meta__item">
          Posted 8 hours 15 minutes ago
        </li>
        <li class="c-bolt-post-info__meta__item">
          Last activity: 2 minutes ago
        </li>
      </ul>
      <div class="c-bolt-post-info__more-info">
        <p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--small">
          This is the more information &hellip; Read More
          <bolt-icon name="chevron-right"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--chevron-right">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   viewbox="0 0 32 32"
              >
                <path fill="var(--bolt-theme-icon, currentColor)"
                      fill-rule="evenodd"
                      d="M13 7c-.6-.5-1.4-.5-2 0-.5.6-.5 1.4 0 2l7.1 7-7 7a1.3 1.3 0 00.9 2.3c.3 0 .7-.1 1-.4l8-8c.5-.5.5-1.3 0-1.8l-8-8z"
                >
                </path>
              </svg>
            </span>
          </bolt-icon>
        </p>
      </div>
      <div class="c-bolt-post-info__warning">
        <p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--small">
          <bolt-icon name="warning"
                     color="yellow"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--warning  c-bolt-icon--yellow">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   viewbox="0 0 24 24"
              >
                <g fill="none"
                   fill-rule="evenodd"
                >
                  <path>
                  </path>
                  <g fill="var(--bolt-theme-icon, currentColor)">
                    <path d="M20.9 20H3a1 1 0 01-.9-1.6l9-14.8a1 1 0 011.7 0l8.9 14.8a1 1 0 01-1 1.6zm2.7-2.7l-9-14.8a3.1 3.1 0 00-5.3 0L.4 17.3a3.1 3.1 0 002.7 4.8H21a3.2 3.2 0 002.7-4.8zM12 7.4a1 1 0 00-1 1v4.2c0 .6.4 1 1 1s1-.4 1-1V8.4c0-.5-.4-1-1-1M12 15.8a1 1 0 100 2 1 1 0 000-2">
                    </path>
                  </g>
                </g>
              </svg>
            </span>
          </bolt-icon>
          This is a warning message
        </p>
      </div>
    </div>
    <ul class="c-bolt-post-status">
      <li class="c-bolt-post-status__item">
        <span class="c-bolt-post-status__icon c-bolt-post-status__icon--solved"
              aria-hidden="true"
        >
          <bolt-icon name="check-solid"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--check-solid">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   data-name="Layer 1"
                   viewbox="0 0 22 22"
              >
                <path fill="var(--bolt-theme-icon, currentColor)"
                      d="M11 0a11 11 0 1011 11A11 11 0 0011 0zm5.2 8.3L10 14.4a1 1 0 01-1.5 0l-2.8-2.8a1 1 0 010-1.5 1 1 0 011.5 0l2 2 5.4-5.3a1 1 0 01.7-.3 1 1 0 01.8.3 1 1 0 010 1.5z"
                >
                </path>
              </svg>
            </span>
          </bolt-icon>
        </span>
        custom solved label
      </li>
      <li class="c-bolt-post-status__item">
        <span class="c-bolt-post-status__icon c-bolt-post-status__icon--locked"
              aria-hidden="true"
        >
          <bolt-icon name="lock"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--lock">
              <svg class="c-bolt-icon__icon"
                   xmlns="http://www.w3.org/2000/svg"
                   viewbox="0 0 24 24"
              >
                <g fill="none"
                   fill-rule="evenodd"
                >
                  <path>
                  </path>
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        d="M20 20c0 .6-.4 1-1 1H5a1 1 0 01-1-1v-7c0-.6.5-1 1-1h14c.6 0 1 .4 1 1v7zM8 7a4 4 0 018 0v3H8V7zm11 3h-1V7A6 6 0 006 7v3H5a3 3 0 00-3 3v7a3 3 0 003 3h14a3 3 0 003-3v-7a3 3 0 00-3-3z"
                  >
                  </path>
                </g>
              </svg>
            </span>
          </bolt-icon>
        </span>
        custom locked label
      </li>
      <li class="c-bolt-post-status__item">
        <span class="c-bolt-post-status__number">
          33
        </span>
        Replies
      </li>
      <li class="c-bolt-post-status__item">
        <span class="c-bolt-post-status__number">
          123
        </span>
        Views
      </li>
    </ul>
  </div>
  <div class="c-bolt-post__actions">
    <bolt-popover placement="bottom-end"
                  spacing="none"
                  boundary=".js-test-target"
                  uuid="12345"
    >
      <div class="c-bolt-popover c-bolt-popover--bottom c-bolt-popover--text-wrap ">
        <span id="js-bolt-popover-trigger-12345">
          <a href="#js-bolt-popover-12345"
             class="c-bolt-popover__nojs-trigger"
             tabindex="-1"
          >
            <ssr-keep for="bolt-popover">
              <bolt-button size="small"
                           color="tertiary"
                           width="auto"
                           border-radius="regular"
                           align="center"
                           transform="none"
                           icon-only
              >
                <button class="c-bolt-button c-bolt-button--small c-bolt-button--tertiary c-bolt-button--border-radius-regular c-bolt-button--center c-bolt-button--icon-only"
                        is="shadow-root"
                >
                  <ssr-keep for="bolt-button"
                            class="c-bolt-button__icon c-bolt-button__icon-sizer"
                  >
                    <bolt-icon slot="before"
                               name="more"
                               aria-hidden="true"
                    >
                      <span class="c-bolt-icon c-bolt-icon--more">
                        <svg class="c-bolt-icon__icon"
                             xmlns="http://www.w3.org/2000/svg"
                             viewbox="0 0 24 24"
                        >
                          <g fill="none"
                             fill-rule="evenodd"
                          >
                            <path>
                            </path>
                            <g fill="var(--bolt-theme-icon, currentColor)">
                              <path d="M14 12a2 2 0 11-4 0 2 2 0 014 0M6 12a2 2 0 11-4 0 2 2 0 014 0M22.1 12a2 2 0 11-4 0 2 2 0 014 0">
                              </path>
                            </g>
                          </g>
                        </svg>
                      </span>
                    </bolt-icon>
                  </ssr-keep>
                  <ssr-keep for="bolt-button"
                            class="c-bolt-button__item"
                  >
                    Action menu
                  </ssr-keep>
                </button>
              </bolt-button>
            </ssr-keep>
          </a>
        </span>
        <span class="c-bolt-popover__no-js-content"
              id="js-bolt-popover-12345"
        >
          <div class="c-bolt-popover__content">
            <span class="c-bolt-popover__bubble ">
              <a href="#js-bolt-popover-trigger-12345"
                 class="c-bolt-popover__nojs-close"
              >
                <span aria-hidden="true">
                  &times;
                </span>
                <span class="u-bolt-visuallyhidden">
                  Close popover
                </span>
              </a>
              <ssr-keep for="bolt-popover">
                <span slot="content">
                  <bolt-menu role="menu">
                    <div role="presentation"
                         class="c-bolt-menu c-bolt-menu--spacing-small"
                    >
                      <ssr-keep for="bolt-menu"
                                class="c-bolt-menu__title c-bolt-menu__title--spacing-small"
                      >
                        <span slot="title">
                          More actions
                        </span>
                      </ssr-keep>
                      <ssr-keep for="bolt-menu"
                                role="presentation"
                      >
                        <bolt-menu-item role="presentation">
                          <bolt-trigger role="menuitem"
                                        display="block"
                                        no-outline
                          >
                            <button type="button"
                                    class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                                    is="shadow-root"
                            >
                              <ssr-keep for="bolt-menu-item"
                                        role="presentation"
                                        class="c-bolt-menu-item"
                              >
                                <replace-with-children class="c-bolt-menu-item__icon-before"
                                                       aria-hidden="true"
                                >
                                  <span slot="icon-before">
                                    <bolt-icon name="reply"
                                               size="small"
                                               aria-hidden="true"
                                    >
                                      <span class="c-bolt-icon c-bolt-icon--reply c-bolt-icon--small">
                                        <svg class="c-bolt-icon__icon c-bolt-icon__icon--small"
                                             xmlns="http://www.w3.org/2000/svg"
                                             baseprofile="full"
                                             viewbox="0 0 24 24"
                                        >
                                          <path fill="var(--bolt-theme-icon, currentColor)"
                                                d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
                                          >
                                          </path>
                                        </svg>
                                      </span>
                                    </bolt-icon>
                                  </span>
                                </replace-with-children>
                                Reply
                              </ssr-keep>
                            </button>
                          </bolt-trigger>
                        </bolt-menu-item>
                        <bolt-menu-item role="presentation">
                          <bolt-trigger role="menuitem"
                                        display="block"
                                        no-outline
                          >
                            <button type="button"
                                    class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                                    is="shadow-root"
                            >
                              <ssr-keep for="bolt-menu-item"
                                        role="presentation"
                                        class="c-bolt-menu-item"
                              >
                                <replace-with-children class="c-bolt-menu-item__icon-before"
                                                       aria-hidden="true"
                                >
                                  <span slot="icon-before">
                                    <bolt-icon name="heart"
                                               size="small"
                                               color="pink"
                                               aria-hidden="true"
                                    >
                                      <span class="c-bolt-icon c-bolt-icon--heart c-bolt-icon--small c-bolt-icon--pink">
                                        <svg class="c-bolt-icon__icon c-bolt-icon__icon--small"
                                             xmlns="http://www.w3.org/2000/svg"
                                             baseprofile="full"
                                             viewbox="0 0 24 24"
                                        >
                                          <path fill="var(--bolt-theme-icon, currentColor)"
                                                d="M12 21.3L10.6 20C5.4 15.4 2 12.3 2 8.5 2 5.5 4.4 3 7.5 3A6 6 0 0112 5a6 6 0 014.5-2c3 0 5.5 2.4 5.5 5.5 0 3.8-3.4 6.9-8.6 11.5L12 21.3z"
                                          >
                                          </path>
                                        </svg>
                                      </span>
                                    </bolt-icon>
                                  </span>
                                </replace-with-children>
                                Remove from favorites
                              </ssr-keep>
                            </button>
                          </bolt-trigger>
                        </bolt-menu-item>
                        <bolt-menu-item role="presentation">
                          <bolt-trigger role="menuitem"
                                        display="block"
                                        no-outline
                          >
                            <button type="button"
                                    class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                                    is="shadow-root"
                            >
                              <ssr-keep for="bolt-menu-item"
                                        role="presentation"
                                        class="c-bolt-menu-item"
                              >
                                <replace-with-children class="c-bolt-menu-item__icon-before"
                                                       aria-hidden="true"
                                >
                                  <span slot="icon-before">
                                    <bolt-icon name="eye-off"
                                               size="small"
                                               aria-hidden="true"
                                    >
                                      <span class="c-bolt-icon c-bolt-icon--eye-off c-bolt-icon--small">
                                        <svg class="c-bolt-icon__icon c-bolt-icon__icon--small"
                                             xmlns="http://www.w3.org/2000/svg"
                                             viewbox="0 0 24 24"
                                        >
                                          <g fill="none"
                                             fill-rule="evenodd"
                                          >
                                            <path>
                                            </path>
                                            <g fill="var(--bolt-theme-icon, currentColor)">
                                              <path d="M10.1 4.7l1.9-.2c2.6 0 5 1.2 7.4 3.6 1.2 1.4 2 2.7 2.5 3.4L20 14a1 1 0 000 1.4l.7.2c.3 0 .6-.1.8-.4a20 20 0 002.3-3.3 1 1 0 000-1s-1.1-2.1-3-4.2A12.2 12.2 0 0012 2.5c-.8 0-1.6 0-2.3.3A1 1 0 008.9 4c.2.5.7.8 1.2.7zM12 18.5c-2.6 0-5-1.2-7.4-3.7a18.7 18.7 0 01-2.5-3.3c1-1.8 2.4-3.3 3.9-4.6l2.5 2.5A4 4 0 0014 15l2.4 2.4a9 9 0 01-4.4 1.2zm-2-7.6l2.6 2.6A2 2 0 0110 11zm13.2 10.4l-4.5-4.6L2.1.3A1 1 0 00.8.3a1 1 0 000 1.4l3.8 3.8A19.7 19.7 0 000 11a1 1 0 000 1s1 2 3 4.2c2.7 2.8 5.8 4.3 9 4.3 2 0 4-.6 5.7-1.8l4 4a1 1 0 001.4 0c.4-.4.4-1 0-1.4z">
                                              </path>
                                            </g>
                                          </g>
                                        </svg>
                                      </span>
                                    </bolt-icon>
                                  </span>
                                </replace-with-children>
                                Unsubscribed
                              </ssr-keep>
                            </button>
                          </bolt-trigger>
                        </bolt-menu-item>
                      </ssr-keep>
                    </div>
                  </bolt-menu>
                </span>
              </ssr-keep>
            </span>
          </div>
        </span>
      </div>
    </bolt-popover>
  </div>
</article>
`;
